request-example.go

```diff
@@ -584,7 +585,24 @@ func (f *memFile) Mode() os.FileMode {
 func (f *memFile) ModTime() time.Time { return f.modtime }
 func (f *memFile) IsDir() bool        { return f.isdir }
 func (f *memFile) Sys() interface{} {
-	return fakeFileInfoSys()
+	var attrs sshfx.Attributes
+
+	attrs.SetSize(uint64(f.size()))
+	attrs.SetUIDGID(65534, 65534)
+
+	switch {
+	case f.isdir:
+		attrs.SetPermissions(0755 | sshfx.ModeDir)
+	case f.symlink != "":
+		attrs.SetPermissions(0777 | sshfx.ModeSymlink)
+	default:
+		attrs.SetPermissions(0644 | sshfx.ModeRegular)
+	}
+
+	mtime := f.modtime.Unix()
+	attrs.SetACModTime(uint32(mtime), uint32(mtime))
+
+	return &attrs
```

Comment:
Aaaaahhhhh yeah, including a `sshfx.Attributes` as a `Sys()` to expose the ability to fake permissions and UID/GID, etc even on systems that do not support those in their native file stat or dirent objects.